src/main/java/com/zachary_moore/runner/LintRunner.java

```diff
@@ -84,25 +92,38 @@ public LintRunner(LintRegister lintRegister,
         }
         this.lintOutput = lintOutput;
 
-        return lintOutput;
+        analyzeLintAndSetExitCode();
     }
 
     /**
      * Check lint output results and return proper exit code
      * @return 0 if no {@link LintRule} report errors and are set to {@link LintLevel#ERROR} else 1
      */
-    public int analyzeLintAndGiveExitCode() {
+    private void analyzeLintAndSetExitCode() {
         if (this.lintOutput == null) {
             throw new RuntimeException("Attempted to analyze lint results before they were computed.");
         }
 
-        for (Map.Entry<LintRule, Map<JSONFile, List<String>>> entry: this.lintOutput.entrySet()) {
+        for (Map.Entry<LintRule, Map<JSONFile, List<Error>>> entry: this.lintOutput.entrySet()) {
             if (entry.getKey().getLevel() == LintLevel.ERROR
                     && entry.getValue().size() != 0) {
-                return 1;
+                exitCode = 1;
+                return;
             }
         }
 
-        return 0;
+        exitCode = 0;
+    }
+
+    public Map<LintRule, Map<JSONFile, List<Error>>> getLintOutput() {
```

Comment:
Can we refactor this to be a stateless object? I'm not sure why I made it hold onto output originally but it seems better to have it be stateless

Reply:
Currently its difficult to make it stateless. The output of the `lint` is `Map<LintRule, Map<JSONFile, List<Error>>>`, `invalid files`, and `exit code`. We cant change the output format as its tightly coupled with `ReportGenerator`.  Once we move to the templatised engine, we can then refractor this file